pkgs/by-name/wa/waywall/package.nix

```
@@ -0,0 +1,79 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  egl-wayland,
+  libglvnd,
+  libspng,
+  libxkbcommon,
+  luajit,
+  mesa_glu,
+  meson,
+  ninja,
+  pkg-config,
+  wayland,
+  wayland-protocols,
+  wayland-scanner,
+  xorg,
+  xwayland,
+}:
+stdenv.mkDerivation {
+  pname = "waywall";
+  version = "0-unstable-2025-02-07";
+
+  src = fetchFromGitHub {
+    owner = "tesselslate";
+    repo = "waywall";
+    rev = "be96e20";
+    hash = "sha256-77GbBzHjyZuauhl0vlguUS/7jBT4qNjOLYGWBVTPjEY=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    meson
+    ninja
+    pkg-config
+    wayland-scanner
+  ];
+
+  buildInputs = [
+    egl-wayland
+    libglvnd
+    libspng
+    libxkbcommon
+    luajit
+    mesa_glu
+    meson
+    ninja
+    pkg-config
+    wayland
+    wayland-protocols
+    xorg.libxcb
+    xwayland
+  ];
+
+  dontUseCmakeConfigure = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp waywall/waywall $out/bin
+  '';
+
+  meta = with lib; {
+    description = "Wayland compositor for Minecraft speedrunning";
+    longDescription = ''
+      Waywall is a Wayland compositor that provides various convenient
+      features (key rebinding, Ninjabrain Bot support, etc) for Minecraft
+      speedrunning. It is designed to be nested within an existing Wayland
+      session and is intended as a successor to resetti.
+    '';
+    homepage = "https://tesselslate.github.io/waywall/";
+    license = licenses.gpl3;
```

```suggestion
    license = lib.licenses.gpl3Only;
```
https://github.com/tesselslate/waywall#license